statsd/client.go

```diff
@@ -304,13 +303,24 @@ func NewClient(addr, prefix string) (Statter, error) {
 	return client, nil
 }
 
-// helper method to ensure a dot is added only when necessary
-func dotprefix(prefix string) string {
-	if prefix != "" && !strings.HasPrefix(prefix, ".") {
-		return "." + prefix
+// joinscopes is a helper that ensures we combine scopes with a dot when
+// it's appropriate to do so. Specifically this looks for trailing dots on
+// the existing prefix and leading dots on the new suffix.
+//
+// It returns the new prefix.
+func joinscopes(curPrefix, suffix string) string {
+	var (
+		emptyBase      = curPrefix == ""
+		baseHasDot     = strings.HasSuffix(curPrefix, ".")
```

Comment:
Yea, that was a bit of "double check all the things;" your suggested take seems reasonable.

:white_check_mark: